tests/umockautoignoreargs_ut/umockautoignoreargs_ut.c

```diff
@@ -322,4 +322,106 @@ TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_for_2nd_arg_when_firs
     ASSERT_ARE_EQUAL(int, 1, is_ignored);
 }
 
+/* Tests_SRS_UMOCKAUTOIGNOREARGS_01_010: [ umockautoignoreargs_is_call_argument_ignored shall look for the arguments as being the string between the last set of parenthesis in call. ]*/
+TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_for_ignored_ptr_arg_when_other_parens_are_present_in_function_call)
+{
+    // arrange
+    int result;
+    int is_ignored;
+
+    // act
+    result = umockautoignoreargs_is_call_argument_ignored("WRAPPER(a)(IGNORED_PTR_ARG)", 1, &is_ignored);
+
+    // assert
+    ASSERT_ARE_EQUAL(int, 0, result);
+    ASSERT_ARE_EQUAL(int, 1, is_ignored);
+}
+
+/* Tests_SRS_UMOCKAUTOIGNOREARGS_01_010: [ umockautoignoreargs_is_call_argument_ignored shall look for the arguments as being the string between the last set of parenthesis in call. ]*/
+TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_for_ignored_num_arg_when_other_parens_are_present_in_function_call)
+{
+    // arrange
+    int result;
+    int is_ignored;
+
+    // act
+    result = umockautoignoreargs_is_call_argument_ignored("WRAPPER(a)(IGNORED_NUM_ARG)", 1, &is_ignored);
+
+    // assert
+    ASSERT_ARE_EQUAL(int, 0, result);
+    ASSERT_ARE_EQUAL(int, 1, is_ignored);
+}
+
+/* Tests_SRS_UMOCKAUTOIGNOREARGS_01_010: [ umockautoignoreargs_is_call_argument_ignored shall look for the arguments as being the string between the last set of parenthesis in call. ]*/
+TEST_FUNCTION(umockautoignoreargs_is_call_argument_ignored_when_RPAREN_at_end_fails)
+{
+    // arrange
+    int result;
+    int is_ignored;
+
+    // act
+    result = umockautoignoreargs_is_call_argument_ignored("WRAPPER(a)(IGNORED_NUM_ARG(", 1, &is_ignored);
```

Comment:
>"WRAPPER(a)(IGNORED_NUM_ARG(" [](start = 58, length = 29)

seems to be missing tests with extra RPAREN like:

`("WRAPPER(a)(IGNORED_NUM_ARG))", 1, &is_ignored);`


Or

`("foo(IGNORED_NUM_ARG(", 1, &is_ignored);`

Reply:
Added one test. Note that the compiler will still validate these calls.

---
In reply to: [375595618](https://github.com/Azure/umock-c/pull/140#discussion_r375595618) [](ancestors = 375595618)